### PR TITLE
feat: add multichain block clock

### PR DESCRIPTION
**Pull request by axiomatic-aardvark**
Adds `BlockClock`, `Network` (structs) and `NetworkName` (enum) in order for us to have better typing around the multichain elements.

Partially advances https://github.com/graphops/graphcast-sdk/issues/88 

Next steps: 
1. Move parts that can be generalised into the SDK
2. Potentially move `BlockClock` into `Network` as per suggestion here https://github.com/graphops/poi-radio/pull/30#discussion_r1106324200
3. Improve error handling, we still have a ton of `unwrap`s in `main.rs`

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -51,25 +52,115 @@ impl RadioPayloadMessage {
...
-    .iter()
-    .cloned()
-    .collect();
+#[derive(Debug, Clone)]
+pub struct Network {
+    pub name: NetworkName,
+    pub interval: u64,
+}
+
+pub struct BlockClock {
```
Comment (hopeyen) [inline src/lib.rs:61]: Later on maybe we can add `BlockClock` as a mutable field in the `Network` struct

File: src/main.rs [hunk 2]
```diff
@@ -133,88 +132,114 @@ async fn main() {
...
+                partial!( query_graph_node_poi => graph_node_endpoint.clone(), id.clone(), _, _);
+
             // Send POI message at a fixed frequency
-            let examination_frequency = SUPPORTED_NETWORK_INTERVALS.get(&network[..]).expect("Subgraph is indexing an unsupported network, please report an issue on https://github.com/graphops/graphcast-rs");
+            let examination_frequency = NETWORKS
+                .iter()
+                .find(|n| n.name.to_string() == network_name.to_string())
+                .map(|n| n.interval)
+                .unwrap_or_else(|| {
+                    panic!(
```
Comment (hopeyen) [inline src/main.rs:231]: Maybe we should just use `warn!` so that the rest of networks can still be compared

**Review by hopeyen — APPROVED** [review]

File: src/main.rs (hunk 2 above)
Comment (axiomatic-aardvark) [reply src/main.rs:231]: oh yeah, I thought I already removed all the panics :derp:

**Review by hopeyen — APPROVED** [review]

**Comment by hopeyen** [comment]
The tests _should_ pass if the branch gets rebased to main after merging #31

**Comment by axiomatic-aardvark** [comment]
I'll also try to refactor to match changes in https://github.com/graphops/graphcast-sdk/pull/86

**Comment by axiomatic-aardvark** [comment]
> The tests _should_ pass if the branch gets rebased to main after merging #31

awesome, this PR should now be up to date and compatible with all 4 other branches - https://github.com/graphops/poi-radio/pull/22, https://github.com/graphops/poi-radio/pull/31, `main` here and `main` on the SDK (0.0.5). Tests are passing, everything looking smooth.

Lmk the sequence you'd prefer to merge them in 👀

**Review by hopeyen — APPROVED** [review]
great, I think the sequence would be the most clean with #31 merged first, #30 merge into #21, then merge #21 🤔

**Comment by axiomatic-aardvark** [comment]
> great, I think the sequence would be the most clean with #31 merged first, #30 merge into #21, then merge #21 🤔

You mean #22 right, not #21 ?

**Review by hopeyen — APPROVED** [review]

**Review by hopeyen — APPROVED** [review]

**Pull request merged**